subprojects/core/src/main/java/org/gradle/api/internal/project/DefaultCrossProjectModelAccess.java

```diff
@@ -55,22 +62,29 @@ public Map<String, Project> getChildProjects(ProjectInternal referrer, ProjectIn
         return relativeTo.getChildProjectsUnchecked().entrySet().stream().collect(
             Collectors.toMap(
                 Map.Entry::getKey,
-                entry -> LifecycleAwareProject.from((ProjectInternal) entry.getValue(), referrer, instantiator)
+                entry -> LifecycleAwareProject.from((ProjectInternal) entry.getValue(), referrer, eagerLifecycleExecutor, instantiator)
             )
         );
     }
 
     @Override
     public Set<? extends ProjectInternal> getSubprojects(ProjectInternal referrer, ProjectInternal relativeTo) {
         return projectRegistry.getSubProjects(relativeTo.getPath()).stream()
-            .map(project -> LifecycleAwareProject.from(project, referrer, instantiator))
+            .map(project -> LifecycleAwareProject.from(project, referrer, eagerLifecycleExecutor, instantiator))
             .collect(Collectors.toCollection(TreeSet::new));
     }
 
     @Override
     public Set<? extends ProjectInternal> getAllprojects(ProjectInternal referrer, ProjectInternal relativeTo) {
         return projectRegistry.getAllProjects(relativeTo.getPath()).stream()
-            .map(project -> LifecycleAwareProject.from(project, referrer, instantiator))
+            .map(project -> LifecycleAwareProject.from(project, referrer, eagerLifecycleExecutor, instantiator))
+            .collect(Collectors.toCollection(TreeSet::new));
+    }
+
+    @Override
+    public Set<? extends ProjectInternal> getAllprojectsForGradle(ProjectInternal rootProject) {
+        return projectRegistry.getAllProjects(rootProject.getPath()).stream()
+            .map(project -> instantiator.newInstance(LifecycleAwareProject.class, project, rootProject, eagerLifecycleExecutor))
```

Comment:
❓  Is it different from the `getAllProject(referrer=rootProject)`? Why do we need a separate method for this?

Reply:
@alllex This code was changed by force push(sorry, did it before seen your comments), PTAL at new implemenation.